website/docs/tutorials/08-interacting-with-starknet/scripting.md

```diff
@@ -0,0 +1,194 @@
+
+#### Make sure the contract is correct
+
+```shell
+protostar build
+protostar test
+```
+
+In order to be able to use this, you have to have protostar installed. You can see how to do this [here](../02-installation.md).
+
+These two instructions assure that the contract builds properly and all tests pass.
+
+#### Declare and deploy the contract
+
+Now, we need to first [declare](./03-declare.md) the contract and then [deploy](./04-deploy.md) it.
+
+Normally, we would start with something like this:
+
+```shell
+protostar declare ./build/main.json
+```
+
+But `deploy` command needs the contract's class hash that comes from the `declare`'s command output. Therefore we need to get this output in a standardized way. That's when the `--json` flag comes into play.
+
+By doing:
+
+```shell
+protostar declare ./build/main.json --json
+```
+
+we get an output like this:
+
+```json
+{"class_hash":"0x07a70656b5612a2f87bd98af477c0be5fa2113d13fe1069e55ad326a3e6f4fe6","transaction_hash":"0x01f6a2c391d1bd0a51322ba73037ada20e0b30da8232bb86028f813a0d4c1fdb"} 
+```
+
+Now, we can parse the json and pull all the desired information from it easily as json is a format that is widely supported.
+
+We could do something like this in our bash script:
+
+```shell
+OUTPUT=$(protostar declare ./build/main.json --json)
+CLASS_HASH=$(python -c "import sys, json; print(json.loads(sys.argv[1])['class_hash'])" $OUTPUT)
+protostar deploy $CLASS_HASH --inputs 100
+```
+
+#### Call the contract
+
+Now, let's say we want to call our contract. In this case, we need the contract address that is being returned from the `deploy` command call.
+
+We are basically going to do the same thing as previously to pass the contract address from `deploy` to `call`:
+
+```shell
+OUT=$(protostar deploy $CLASS_HASH --inputs 100 --json)
+CONTRACT_ADDRESS=$(python -c "import sys, json; print(json.loads(sys.argv[1])['contract_address'])" $OUT)
+protostar call --contract-address $CONTRACT_ADDRESS --function get_balance
+```
+
+In the end, our bash script looks like this:
+
+```shell
+#!/bin/bash
+
+set -e
+
+protostar build
+protostar test
+
+DECLARE_OUTPUT=$(protostar declare ./build/main.json --json)
+CLASS_HASH=$(python -c "import sys, json; print(json.loads(sys.argv[1])['class_hash'])" $DECLARE_OUTPUT)
+
+DEPLOY_OUTPUT=$(protostar deploy $CLASS_HASH --inputs 100 --json)
+CONTRACT_ADDRESS=$(python -c "import sys, json; print(json.loads(sys.argv[1])['contract_address'])" $DEPLOY_OUTPUT)
+protostar call --contract-address $CONTRACT_ADDRESS --function get_balance
+```
+
+Of course, we could exclude pulling a specific field from the output to a separate bash function but that's not important.
+
+Using `--json` flag is also good if you prefer the more compressed output from protostar commands.
```

Comment:
If this file is ok here, it might be a good idea to prepare a repository and paste a link here